Automattic/wp-calypso
# Composite checkout: fix purchase link in duplicate notice

#### Changes proposed in this Pull Request

When purchasing a Jetpack plan that includes a product you already own, a warning notice is shown, which contains a link to the specific purchase page. Because the composite checkout didn't query the site purchases, that link didn't point to this page but to the purchase list instead.

#### Testing instructions

1. Download the PR
2. Select a self-hosted Jetpack site that has a Backup product and no plan
3. Go to the _Plans_ section
4. Select the Professional plan
5. You should land in the checkout page and see the warning notice. To show the composite checkout, append `?flags=composite-checkout-force` to the URL and reload the page
6. Click both links in the notice (they point to the same URL) and check that you're redirected to `/me/purchases/:site/:purchase-id/` and not `/me/purchases/`

#### Screenshots

Duplicate notice with composite checkout
<img width="942" alt="Screen Shot 2020-07-30 at 10 09 39 AM" src="https://user-images.githubusercontent.com/1620183/88934213-572bed00-d24e-11ea-9b6c-0bbeef8dc521.png">
